azure-iot-device/azure/iot/device/provisioning/aio/async_x509_provisioning_device_client.py

```diff
@@ -0,0 +1,68 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains user-facing asynchronous clients for the
+Azure Provisioning Device SDK for Python.
+"""
+
+import logging
+from azure.iot.device.common import async_adapter
+from ..abstract_provisioning_device_client import AbstractProvisioningDeviceClient
+from ..abstract_provisioning_device_client import log_on_register_complete
+from ..internal.polling_machine import PollingMachine
+
+logger = logging.getLogger(__name__)
+
+
+class X509ProvisioningDeviceClient(AbstractProvisioningDeviceClient):
```

Comment:
Do we actually need separate device clients for X509 and symmetric key?  So far, I don't see any difference.

Reply:
yes this is the discussion point I brought up in the scrum last time. So far everywhere the client is separate but if the functionality is so similar we can simply have 1 client , or we can have implementations in the abstract client.

---
In reply to: [291683727](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291683727) [](ancestors = 291683727)